chicagorequests/__init__.py

```diff
@@ -178,4 +189,8 @@ def main(
             unit="day",
         ):
             for result in day:
+                if json_file:
+                    json.dump(result, json_file, ensure_ascii=False, indent=2)
```

Comment:
this won't be a valid json file if we have more than one day.

one way to do that without taking a ton of memory would be like this

```python
...
json_file.write('[\n'])
...
for result in day:
    if json_file:
        json_file.write(json.dumps(result, ensure_ascii=False, indent=2) + ',\n')

...
if json_file:
    json_file.write(']')
    json_file.close
```

Reply:
I noticed that on your output. I thought about modifying the structure for something that would be valid json, but if we're not spitting that out as is...what's the point? If it's meant to be human readable, it isn't as is, and since the output can't be easily redirected as something that's valid json to another program anyway, then I figured that was the use case. The file could easily be loaded into blocks and imported as json for those, which is what any program parsing this would have to do anyway.

Otherwise, yes, we have 1) memory issues holding the whole thing before writing, and 2) format issues as the output to stdout is different than what's in the file.